File: ios/CanvasReactNative.m
#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(CanvasReactNative, RCTViewManager)
    RCT_EXTERN_METHOD(sync:(nonnull NSNumber *)node
                      command:(nonnull NSString *) command
                      )
@end

File: ios/CanvasReactNative.swift
import Foundation
import UIKit

@objc(CanvasReactNative)
class CanvasReactNative: RCTViewManager {
    var canvas: Canvas!
    override func view() -> UIView! {
        canvas = Canvas()
        canvas.backgroundColor = UIColor.white
        return canvas
    }

    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

     @objc public func sync(_ node:NSNumber, command:String){
         DispatchQueue.main.sync {
            canvas.sync(command: command)
         }
     }
}
